### ISSUE TITLE
Issue since v.0.14.1 : no logs in output file

### ISSUE THREAD

**Issue by ylegat** [buger/gor#303]
Hey @buger , I think there is an issue with version 0.14.1. Apparently, logs are not written in the output file. Gor is launch with :
```
sudo ./gor --input-raw :8086  --verbose --debug --output-file ./request.gor
```

Two HTTP requests are then executed and we stop Gor manually. After the stop, the log file has been created but it is empty. We did a try using version 0.13.0 using the same protocol and it worked. Of course due to the bug present in the version 0.14.0 we were not able to test it.

Are you aware of this issue ?

**EDIT 1** : the data are correctly written when the flush is called by the go routine, so every minutes by default
**EDIT 2** : actually, we are always missing the first requests logged, even after some flushes. I just did a try without 2000 requests. The first one (sometimes a few, sometimes a hundred) are missing. I think the first file is neither flush nor closed.

**Comment by ylegat** [comment]
I've found the origin of the issue : we output-file we pass is `./request.gor`, but other paths are computed without the `./`, so at one moment we've got `"./request_0.gor" != "request_0.gor"` [here](https://github.com/buger/gor/blob/master/output_file.go#L181) and the file is crushed.

So the fix is simple : filename given with the command should be normalized.

**Comment by mlallaouret** [comment]
Hello @ylegat and @buger 

Indeed the `./`in the ouput-file path causes the loss of the first requests in the file. I will propose a PR to resolve this issue.

But there is still an issue with the flush of the writer when we stop the process. If we do not wait the end of the minute in order to flush the writer with the go routine and that we stop the process, no data is written. Is it intended not to flush the writer at the end of the process since the version 0.14.1 ?

**Comment by mlallaouret** [comment]
I found the issue with the writer that is not flushed at the end of program. In `gor.go`, when the go routine receive the `SIGTERM` signal, gor iterate over all the plugins to call the `Close`method. The problem with `FileOutput.Close`is that the implementation is not correct. It should declare an error as return type.

**Comment by buger** [comment]
Bot issues should be fixed, thanks to @mlallaouret !

**Comment by buger** [comment]
Here are binaries with fixes:
[gor_DEV-1466094930_mac.tar.gz](https://github.com/buger/gor/files/318895/gor_DEV-1466094930_mac.tar.gz)
[gor_DEV-1466094909_x64.tar.gz](https://github.com/buger/gor/files/318894/gor_DEV-1466094909_x64.tar.gz)

**Comment by QasimK** [comment]
@buger Thank you for the fixed binaries. Perhaps there could be a bigger notice for this issue? I spent quite a while tracking down what was going on

**Comment by buger** [comment]
More verbose error messages will definitly help

**Comment by buger** [comment]
@QasimK @ylegat @mlallaouret happy to announce just released  [v0.15.0](https://github.com/buger/gor/releases/tag/v0.15.0) version, which include those fixes.
